drivers/nrf24l01p_ng/nrf24l01p_ng_states.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup drivers_nrf24l01p_ng
+ * @{
+ *
+ * @file
+ * @brief   Implementation of state transition procedures
+ *          for the NRF24L01+ (NG) transceiver
+ *
+ * @author Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+#include <assert.h>
+#include <string.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#include "xtimer.h"
+#include "net/netopt.h"
+
+#include "nrf24l01p_ng_registers.h"
+#include "nrf24l01p_ng_communication.h"
+#include "nrf24l01p_ng_states.h"
+
+static void _restore_address(const nrf24l01p_ng_t *dev)
```

Comment:
What does it mean to restore the address / why is it needed?

Reply:
Any pipe 0 - 5 can be assigned an address to receive data on. Pipe 0 is the transmission pipe. If you want to send something, you override the address of pipe 0 and set it to the destination address. After you got an ACK for the frame or a timeout, you have to restore the listen address of pipe 0.